docs: link directive (#3202)
apollographql/federation: docs/shared/link-directive.mdx

> Subgraph schemas "opt in" to Federation v2 features by applying the `@link` directive to the `schema` type. The `import` list of this definition must include each federation-specific directive that the subgraph schema uses. In the example below, the schema uses the `@key` and `@shareable` directives:

Reviewer: Not sure if `"opt in"` requires the quotes, but I'm a minimalist when it comes to characters like `"` and `(` so take that with a grain of sale.

> Subgraph schemas "opt in" to Federation v2 features by applying the `@link` directive to the `schema` type. The `import` list of this definition must include each federation-specific directive that the subgraph schema uses. In the example below, the schema uses the `@key` and `@shareable` directives:

Reviewer 2: > The `import` list of this definition must include
Imports are optional and it is perfectly fine to skip it. Side effect is that without explicit imports those directives have to be namespaced to the spec (or provided alias), i.e. `@key` becomes `@federation__key` without explicit import

Author: thanks @dariuszkuc, do these edits make sense:

```suggestion
Subgraph schemas opt in to Federation v2 features by applying the `@link` directive to the `schema` type. You can optionally add an `import` list to this definition to include each federation-specific directive that the subgraph schema uses. In the example below, the schema uses the `@key` and `@shareable` directives:
```
